File: src/tools/lizard/eval-repos/synthetic/javascript/edge_cases/deep_nesting.js
/**
 * Deep nesting test - 10+ levels of indentation.
 */

function deeplyNestedFunction(data) {
  let result = '';

  if (data) {
    // Level 1
    if (data.level1) {
      // Level 2
      for (const item of data.level1) {
        // Level 3
        if (typeof item === 'object') {
          // Level 4
          for (const [key, value] of Object.entries(item)) {
            // Level 5
            if (value) {
              // Level 6
              try {
                // Level 7
                if (Array.isArray(value)) {
                  // Level 8
                  for (const v of value) {
                    // Level 9
                    if (v > 0) {
                      // Level 10
                      let temp = v;
                      while (temp > 0) {
                        // Level 11
                        if (temp % 2 === 0) {
                          // Level 12
                          result += temp.toString();
                        }
                        temp--;
                      }
                    }
                  }
                }
              } catch (e) {
                // Ignored
              }
            }
          }
        }
      }
    }
  }

  return result;
}

class DeeplyNestedClass {
  processMatrix(matrix) {
    let total = 0;

    if (matrix) {
      // 1
      for (const row of matrix) {
        // 2
        if (row) {
          // 3
          for (const cell of row) {
            // 4
            if (typeof cell === 'object') {
              // 5
              for (const [k, v] of Object.entries(cell)) {
                // 6
                if (v !== null) {
                  // 7
                  switch (typeof v) {
                    // 8
                    case 'number':
                      if (v > 0) {
                        // 9
                        for (let i = 0; i < v; i++) {
                          // 10
                          if (i % 2 === 1) {
                            // 11
                            total += i;
                          }
                        }
                      }
                      break;
                  }
                }
              }
            }
          }
        }
      }
    }

    return total;
  }
}

module.exports = {
  deeplyNestedFunction,
  DeeplyNestedClass,
};

File: src/tools/lizard/eval-repos/synthetic/rust/edge_cases/deep_nesting.rs
//! Deep nesting test - 10+ levels of indentation.

use std::collections::HashMap;

/// Function with 10+ levels of nesting.
pub fn deeply_nested_function(data: Option<HashMap<String, Vec<HashMap<String, Vec<i32>>>>>) -> String {
    let mut result = String::new();

    if let Some(data) = data {  // Level 1
        if let Some(level1) = data.get("level1") {  // Level 2
            for item in level1 {  // Level 3
                for (key, value) in item {  // Level 4
                    let _ = key;
                    if !value.is_empty() {  // Level 5
                        for n in value {  // Level 6
                            if *n > 0 {  // Level 7
                                let mut temp = *n;
                                while temp > 0 {  // Level 8
                                    if temp % 2 == 0 {  // Level 9
                                        if temp < 100 {  // Level 10
                                            if temp > 0 {  // Level 11
                                                result.push_str(&temp.to_string());
                                            }
                                        }
                                    }
                                    temp -= 1;
                                }
                            }
                        }
                    }
                }
            }
        }
    }

    result
}

/// Matrix processor with deep nesting.
pub fn process_matrix(matrix: Option<Vec<Vec<Vec<HashMap<String, i32>>>>>) -> i32 {
    let mut total = 0;

    if let Some(matrix) = matrix {  // 1
        for row in matrix {  // 2
            if !row.is_empty() {  // 3
                for cell in row {  // 4
                    if !cell.is_empty() {  // 5
                        for item in cell {  // 6
                            for (_, v) in item {  // 7
                                if v > 0 {  // 8
                                    for i in 0..v {  // 9
                                        if i % 2 == 1 {  // 10
                                            if i < 1000 {  // 11
                                                total += i;
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }

    total
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_deeply_nested() {
        let result = deeply_nested_function(None);
        assert!(result.is_empty());
    }
}
